INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (1,'Bluescript with a T7 promoter on each side of the MCS driving transcription of each DNA  strand','Mary Abraham; ttp://www.ciwemb.edu (Nature, 395: 854)','L4440','Amp','DH5-alpha.');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (2,'rnc14::Tn10(DE3 lysogen: lavUV5 promoter -T7 polymerase) (IPTG-inducible T7 polymerase) (RNase III minus) F-, mcrA, mcrB, IN(rrnD-rrnE)1, lambda -,','Mary Abraham; Feeding Strain Tn10 carries Tet(R).','HT115','Tet','HT115');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (3,'Empty L4440 vector  F-, mcrA, mcrB, IN(rrnD-rrnE)1, lambda -, rnc14::Tn10(DE3 lysogen: lavUV5 promoter -T7 polymerase) (IPTG-inducible T7 polymerase) (RNase III minus)','Mary Abraham; Feeding Strain Tn10 carries Tet(R).','HT115','Amp + Tet','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (4,'GFP in L4440 vector  F-, mcrA, mcrB, IN(rrnD-rrnE)1, lambda -, rnc14::Tn10(DE3 lysogen: lavUV5 promoter -T7 polymerase) (IPTG-inducible T7 polymerase) (RNase III minus)','Mary Abraham; Feeding Strain Tn10 carries Tet(R).','HT115','Amp + Tet','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (5,'UNC22 in L4440 vector  F-, mcrA, mcrB, IN(rrnD-rrnE)1, lambda -, rnc14::Tn10(DE3 lysogen: lavUV5 promoter -T7 polymerase) (IPTG-inducible T7 polymerase) (RNase III minus)','Mary Abraham; Feeding Strain Tn10 carries Tet(R).','HT115','Amp + Tet','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (6,'OP50','feeding strain for C.elegans','OP50','None','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (7,'CEN/ARS-LYS2','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS317','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (8,'CEN/ARS-MET15','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS411','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (9,'CEN/ARS-ADE2','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS412','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (10,'CEN/ARS-HIS3','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS413','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (11,'CEN/ARS-TRP1','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS414','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (12,'CEN/ARS-LEU2','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS415','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (13,'CEN/ARS-URA3','ATCC87537 S. cerevisiae/E. coli YC-type pRS shuttle vectors','pRS416','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (14,'MET15, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS401','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (15,'ADE2, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS402','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (16,'HIS3, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS403','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (17,'TRP1, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS404','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (18,'LEU2, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS405','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (19,'URA3, integrating','ATCC87536 S. cerevisiae/E. coli YI-type pRS shuttle vectors','pRS406','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (20,'YFP (Venus), Kan, 4882bp.','Venus (G418 selection). Base on pDH6. http://depts.washington.edu/~yeastrc/fm_home16.htm','pBS7','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (21,'YFP, his5 (S.pombe), 4726bp.','a pH-tolerant YFP (HIS3) http://depts.washington.edu/~yeastrc/fm_home2.htm','pDH5','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (22,'YFP, Kan, 4882bp.','YFP (G418 selection).  Base on pDH5 and pDH3. http://depts.washington.edu/~yeastrc/fm_home13.htm','pDH6','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (23,'CFP, Kan, 4882bp.','CFP (G418 selection).  http://depts.washington.edu/~yeastrc/fm_home1.htm','pDH3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (24,'CFP, hph, 4875bp.','CFP (hygromycin-B selection). http://depts.washington.edu/~yeastrc/fm_home15.htm','pBS4','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (25,'DsRed.T1.N1, Kan, 4846bp.','Feb, 2003','pTY24; DsRed.T1.N1 (G418 selection)','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (26,'pFA6a harboring pTEF-Kan-tTEF for knockout.','RDB903, from Peter Philippsen. http://depts.washington.edu/~yeastrc/fm_home17.htm','pFA6a-KanMX4','Amp.','DH5alpha');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (27,'pFA6a harboring pTEF-(S.pombe his5+)-tTEF for knockout.','RDB904, from Peter Philippsen.','pFA6a-His3Mx6','Amp.','DH5alpha');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (28,'CEN/ARS ADE8 plasmid, 6333bp.','Yeast 2001 Apr;18(6):563-75 steve.oliver@man.ac.uk, d.struttman@bso.man.ac.uk','pRS4110','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (29,'(SacI) ADH promoter (HindIII) SV40-NLS (EcoRI) P12 mutant PKI NES (BamH1) GFPx2 (XhoI) from pKW431was cloned into pRS404.','JBO3T-1, Jamie@ Cross Lab Ref:  Cell 90:1041-1050.','pRS404-pKW431','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (30,'(SacI) ADH promoter (HindIII) SV40-NLS (EcoRI) P12 mutant PKI NES (BamH1) GFPx2 (XhoI) from pKW431was cloned into pRS404.','JBO3T-2, an indept isolate identical to WSB29.','pRS404-pKW431','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (31,'pRS404(SacI) ADH promoter (HindIII) YFP-venus+~10 a.a. (XhoI)','WSB20 +WSOs (13,8) -> YFP- venus; WSB29+30 -> ADHp','60L2#2','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (32,'pRS404(SacI) ADH promoter (HindIII) YFP-venus+~10 a.a. (XhoI)','an indept isolate identical to WSB31.','60L2#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (33,'pRS404(SacI) ADH promoter (HindIII) CFP+~10 a.a. (XhoI)','WSB23 +WSOs (13,8) ->CFP; WSB29+30 -> ADHp','60L3#2','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (34,'pRS404(SacI) ADH promoter (HindIII) CFP+~10 a.a. (XhoI)','an indept isolate identical to WSB33.','60L3#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (35,'pFA6a-kanMX4-yEGFP tagging cassette','Knop et al., Yeast 15:963-972 (1999).  contains S65G, S72A mutations near the chromo- phore; all codons in this version of the GFP gene are optimal for translation in yeast.','pYM12','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (36,'not F\'','from Santiago, frozen by Craig Correll. Can use this to make competent cells','RDB626','DH5alpha','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (37,'pRS404(SacI) ADH promoter (HindIII) YFP-venus (XhoI)','WSB20 +WSOs (13,15) -> YFP- venus; WSB29+30 -> ADHp','62L1#2','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (38,'pRS404(SacI) ADH promoter (HindIII) YFP-venus (XhoI)','an indept isolate identical to WSB37.','62L1#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (39,'pRS404(SacI) ADH promoter (HindIII) CFP (XhoI)','WSB23 +WSOs (13,15) -> CFP; WSB29+30 -> ADHp','62L2#2','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (40,'pRS404(SacI) ADH promoter (HindIII) CFP (XhoI)','an indept isolate identical to WSB39.','62L2#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (41,'pRS404(SacI) ADH promoter (HindIII) RGS-His6-dsRed.T4(XhoI)','WSB47 + WSOs (21,16) -> His6-dsRed; WSB29+30 -> ADHp','83L2#1','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (42,'pRS404(SacI) ADH promoter (HindIII) RGS-His6-dsRed.T4(XhoI)','an indept isolate identical to WSB41.','83L2#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (43,'pRS404(SacI) ADH promoter (HindIII) yEGFP(XhoI)','WSB35 +WSOs (17,18) -> yEGFP; WSB29+30 -> ADHp','62L4#1','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (44,'pRS404(SacI) ADH promoter (HindIII) yEGFP(XhoI)','an indept isolate identical to WSB43.','62L4#3','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (45,'pFA6a-GFP(S65T)-HIS3MX6; HIS5 gene of  S. pombe driven by TEF promoter, GFP(S65T), bla gene, multiple cloning site, pSP72 derivative','Yeast 10: 1793-1808 (1994). peter.philippsen@unibas.ch','pFA6a-GFP(S65T)-HIS3MX6','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (46,'pFA6a-GFP(S65T)-KanMX6; kanamycin and geneticin (G418) resistance gene, bla, GFP(S65T), multiple cloning site, pSP72 derivative.','Yeast 10: 1793-1808 (1994). peter.philippsen@unibas.ch','pFA6a-GFP(S65T)-KanMX6','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (47,'DsRedT4 was cut with Not1, blunted with Klenow, cut with BamHI, and cloned into pQE81-L at BamHI and Ecl136II (GAG/CTC).','low copy #! For IPTG-inducible gene expression in E.coli.  From B. Glick.  20-30 copies of plasmid per bacterium.  pQE81 is from Qiagen, with ColE1 origin of replication.','pQE81-L-DsRed.T4','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (48,'a 335bp fragment containing the lambda attP site w/ NheI and AvrII @ the ends.  This was subcloned into pZE12-GFPaav cut with AvrII.  The resulting plasmid, pZE12Int-GFPaav, has a T1-attP fragment which can be transferred to other pZ vectors with HindIII-AvrII or XbaI-AvrII. For integration, cut out the colE1 origin with SpeI and AvrII, then recircularize (cohesive compatible). The Lac promoter undergoes mutation (due to intrinsic repeats?) when driving toxic protein at very high levels.  Gradually, the level of expression will drop.    For integration into attB site.  Not AAV, but ASV! 3351 bp','John Chuan','pZE12Int-GFPaav','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (49,'a contaminant bacterium; colonies yellowish; capable of weaving yeast co-biofilm.','8=Ade,Ura,Trp,His,Leu, Lys,  Arg,Met','SD+8','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (50,'lys20::TRP1; use Not1/Not1 fragment to transform yeast of trp1 background','plasmid; fanarg@ulb.ac.be Eur.J.Biochem.261,163-170.  Use (WSO32, WSO33) to screen for 670bp.','pLAF109','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (51,'lys21::LEU2; use Not1/Not1 fragment to transform yeast of leu2 background','plasmid; fanarg@ulb.ac.be Eur.J.Biochem.261,163-170.  Use (WSO35, WSO36) to screen for 1.3kb.','pLAF112','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (52,'LYS20-fbr2; use 2.2kb BamH1/BamH1 fragment to transf. lys20::TRP1 lys21::LEU2','plasmid; fanarg@ulb.ac.be Eur.J.Biochem.261,163-170.','pAI8','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (53,'LYS21-fbr; use 1.99kb BamH1/BamH1 fragment to transf. lys20::TRP1 lys21::LEU2','plasmid; fanarg@ulb.ac.be Eur.J.Biochem.261,163-170.','pAI16','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (54,'pRS406-ADHp-NLS-P12mut-GFPGFP','linearize with Stu1 to integrate into ura3-52. P12mut has a mutation in nuclear export signal. Upon transformation, no  detectable GFP, let alone nuclear localized.','WSL2-1','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (55,'pRS406-ADHp-NLS-DsRed','linearize with Stu1 to integrate into ura3-52. p241. Localized to nucleus, but very low expression.','WSL4-1','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (56,'pRS406-ADHp-NLS-DsRed','linearize with Stu1 to integrate into ura3-52. p241.  A different transformant.','WSL4-2','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (57,'MET-dsRed-NLS  URA3 CEN; dsRed-NLS is the Xba1/Xho1 fragment; the entire MET-dsRed-NLS, you could use SacI/KpnI cut.','Ying Lu,  Rodrigues F, J Bacteriology 183:3791-3794','pUR36NLS','Ampr','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (58,'pRS404-ACT1prDsRED','use SnaBI to integrate pJB06T at the TRP1 locus.  ditalis@mail.rockefeller.edu at Fred Cross lab. Di Talia S et al., NATURE 450 :1272-1272','pJB06T','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (59,'pRS406-ACT1prDsRED','use NcoI to integrate pSD02 at the URA3 locus. ditalis@mail.rockefeller.edu at Fred Cross lab. Di Talia S et al., NATURE 450 :1272-1272','pSD02','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (60,'replacement of the URA3 gene on pRS316 with the KANMX cassette','Lazar at Dan Gottschling lab; pLND30; loc: 42-61. PCR amplified the KANMX cassette from pRS400(KAN) using primers pRS_F1 nd pRS_R1. Co-transformed the PCR product and pRS316 into BY4724 and selected for KAN resistant transformants. Few were checked by colony PCR for correct replacement of the URA3 gene on pRS316 with the KANMX cassette. After plasmid rescue into DH5a this clone was confirmed by restriction digest.','pRS31KAN','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (61,'replacement of the URA3 gene on pRS316 with the NATMX cassette.','Lazar at Dan Gottschling lab; pLND31; loc: 42-62. PCR amplified the NATMX cassette from pRS40NAT using primers pRS_F1 nd pRS_R1. Co-transformed the PCR product and pRS316 into BY4724 and selected for NAT resistant transformants. Few were checked by colony PCR for correct replacement of the URA3 gene on pRS316 with the NATMX cassette. After plasmid rescue into DH5a this clone was confirmed by restriction digest.','pRS31NAT','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (62,'replacement of the URA3 gene on pRS316 with the HYGMX cassette.','Lazar at Dan Gottschling lab; pLND32; loc: 42-63. PCR amplified the HYGMX cassette from pRS40HYG using primers pRS_F1 nd pRS_R1. Co-transformed the PCR product and pRS316 into BY4724 and selected for HYG resistant transformants. Few were checked by colony PCR for correct replacement of the URA3 gene on pRS316 with the HYGMX cassette. After plasmid rescue into DH5a this clone was confirmed by restriction digest.','pRS31HYG','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (63,'mCherry-KANMX.   pBS34 is based upon the pFA6-GFPkan (Wach et al) except that GFP is replaced by mCherry (Shaner et al).','from Gottschling lab: 47-14; http://depts.washington.edu/yeastrc/pages/pBS34.html Must thank Tsien/ yeast resource center in publications.','pBS34','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (64,'pFA6a-link-yEGFP-SpHIS5','EUROSCARF; Accession no. P30174; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT128','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (65,'pFA6a-link-yEGFP-KAN','EUROSCARF; Accession no. P30175; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT127','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (66,'pFA6a-link-yESapphire-SpHIS5','EUROSCARF; Accession no. P30176; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT149','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (67,'pFA6a-link-yESapphire-KAN','EUROSCARF; Accession no. P30177; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT150','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (68,'pFA6a-link-yECFP-SpHIS5','EUROSCARF; Accession no. P30181; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT101','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (69,'pFA6a-link-yECFP-KAN','EUROSCARF; Accession no. P30182; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT102','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (70,'pFA6a-link-yECFP-CaURA3','EUROSCARF; Accession no. P30183; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT174','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (71,'pFA6a-link-yEVenus-SpHIS5','EUROSCARF; Accession no. P30184; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT90','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (72,'pFA6a-link-yEVenus-KAN','EUROSCARF; Accession no. P30185; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT103','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (73,'pFA6a-link-yECitrine-SpHIS5','EUROSCARF; Accession no. P30186; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT139','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (74,'pFA6a-link-yECitrine-KAN','EUROSCARF; Accession no. P30187; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT140','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (75,'pFA6a-link-yECitrine-CaURA3','EUROSCARF; Accession no. P30187; http://web.uni-frankfurt.de/fb15/mikro/euroscarf/data/GFPThorn.html','pKT175','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (76,'CFP fused to 5\' region of Kluyveromyces lactis URA3 gene.  For CFP gene tagging. from Gottschling lab: 36-60; Original ref: Methods in Enzymology vol. 350, part B, p.275','pWJ1162','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (77,'CFP fused to 3\' region of Kluyveromyces lactis URA3 gene.  For CFP gene tagging.  From Gottschling lab: 36-61; Original ref: Methods in Enzymology vol. 350, part B, p.275','pWJ1163','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (78,'YFP fused to 5\' region of Kluyveromyces lactis URA3 gene.  For YFP gene tagging.  From Gottschling lab: 36-62; Original ref: Methods in Enzymology vol. 350, part B, p.275','pWJ1164','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (79,'YFP fused to 3\' region of Kluyveromyces lactis URA3 gene.  For YFP gene tagging.  From Gottschling lab 36-63; Original ref: Methods in Enzymology vol. 350, part B, p.275','pWJ1165','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (80,'mDsRed fused to 5\' region of Kluyveromyces lactis URA3 gene. For RFP gene tagging.  From Rothstein Lab.  Modified version of plasmids in from Methods in Enzymology vol. 350, part B, p.275','pWJ1350','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (81,'mDsRed fused to 5\' region of Kluyveromyces lactis URA3 gene. For RFP gene tagging.  From Rothstein Lab.  Modified version of plasmids in Methods in Enzymology vol. 350, part B, p.275','pWJ1351','Amp','','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (82,'pUC19-YFP-CDC10-LEU2','YFP-CDC10 fusion with LEU2 marker.  From Kyung Lee (kyunglee@mail.nih.gov).','pKL935','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (83,'pUC19-MYO1-GFP-LEU2','MYO1-GFP fusion with LEU2 marker.  From Kyung Lee (kyunglee@mail.nih.gov).','pKL937','Amp','');

INSERT INTO bacterial_plasmids (number,description,source,plasmid_name,drug_resistance,background)
VALUES (84,'Designer deletion plasmid for LYS2.  Linearize with ClaI.  Select for Ura+.  Pop out Ura by growing on 5-Foa.','pAD2','Amp','','');

